src/js/pages/GranthIndex/components/GranthIndex.tsx

```diff
@@ -1,38 +1,150 @@
-import React, { useEffect, MouseEvent } from 'react';
+import React, { useEffect, useState, useCallback } from 'react';
 import { Link } from 'react-router-dom';
 import { Location } from 'history';
 
 import BreadCrumb from '@/components/Breadcrumb';
 import Fetch from '@/components/Fetch';
-import { raagIndices } from '../constants/raag-indices';
+import { raagIndices, IRaagIndex } from '../constants/raag-indices';
 import { toAngURL } from '../../../util';
 import { pageView } from '../../../util/analytics';
 import { AmritKeertanIndex } from './AmritKeertanIndex';
-import { sanitizeHash, scrollToHash } from '../util';
+import { sanitizeHash, scrollToHash, desanitzeHash } from '../util';
 import { TEXTS } from '../../../constants';
 
 interface IGranthIndexProps {
-  location: Location
+  location: Location;
 }
 
+const GRANTH_NAMES = raagIndices.map(({ name }) => name);
+
 export const GranthIndex: React.FC<IGranthIndexProps> = ({ location }) => {
   const { hash } = location;
 
+  const [selection, setSelection] = useState<string[]>(() =>
+    hash ? desanitzeHash(hash) : [GRANTH_NAMES[0]]
+  );
+
+  // Helps check if a list item is selected to show subSets and description
+  const isSelected = useCallback(
+    (name: string) => {
+      if (selection.includes(name)) {
```

Comment:
@yudhv prolly we can directly return `selection.includes(name)` it gonna return boolean.